### PR TITLE
Refactor tests & fix some declarations

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
This should significantly improve the UX when designing tests, especially since we will add more configuration options for the upcoming byzantine tests.

### CODE REVIEW

File: source/agora/test/Base.d [hunk 1]
```diff
@@ -570,23 +577,15 @@ public enum NetworkTopology
...
         The set of public key added to the node
 
 *******************************************************************************/
 
-public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)
-    (NetworkTopology topology, size_t nodes, bool configure_network = true,
-        long retry_delay = 100, size_t max_retries = 20, long timeout = 500,
-        size_t max_failed_requests = 100)
+public APIManager makeTestNetwork (APIManager : TestAPIManager = TestAPIManager)(
+    in TestConf test_conf = TestConf.init)
```
Comment (AndrejMitrovic) [inline source/agora/test/Base.d]: Actually, I changed my mind. I don't want a default value here. I just got bitten by it lol.

File: source/agora/test/Simple.d [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*******************************************************************************
+
+    Contains the simplest possible block creating test
+
```
Comment (Geod24) [inline source/agora/test/Simple.d:9]: Add the commit description (especially how to run it) here or in README of the test directory.

File: source/agora/test/Simple.d [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
```
Comment (Geod24) [inline source/agora/test/Simple.d]: C U R R E N T Y E A R

File: source/agora/test/Base.d [hunk 4]
```diff
@@ -542,6 +542,31 @@ public enum NetworkTopology
     SinglePoF,
 }
 
+/// Node / Network / Quorum configuration for use with makeTestNetwork
+struct TestConf
```
Comment (Geod24) [inline source/agora/test/Base.d]:
```suggestion
public struct TestConf
```

**Review by Geod24 — APPROVED** [review]
Noice

**Comment by AndrejMitrovic** [comment]
Thank you for the review. Updated.

**Comment by AndrejMitrovic** [comment]
This was approved, merging.

**Pull request merged**